src/prommis/solvent_extraction/tests/test_sx_model_dynamic.py

```diff
@@ -0,0 +1,261 @@
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Ca"].fix(4.955e-9)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Fe"].fix(1.491e-7)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Sc"].fix(321.34)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Y"].fix(5.67e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["La"].fix(1.78e-05)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Ce"].fix(4.019e-5)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Pr"].fix(6.73e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Nd"].fix(1.82e-5)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Sm"].fix(3.285e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Gd"].fix(1.55e-6)
+        m.fs.solex.mscontactor.organic_inlet_state[:].conc_mass_comp["Dy"].fix(9e-7)
+
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Al"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ca"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Fe"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sc"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Y"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["La"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Ce"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Pr"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Nd"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Sm"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Gd"].fix(1e-9)
+        m.fs.solex.mscontactor.organic[0, :].conc_mass_comp["Dy"].fix(1e-9)
+
+        m.fs.solex.mscontactor.organic_inlet_state[:].flow_vol.fix(62.01)
+
+        m.fs.solex.mscontactor.organic[0, :].flow_vol.fix(62.01)
+
+        return m
+
+    @pytest.mark.component
+    def test_structural_issues(self, SolEx_frame):
+        model = SolEx_frame
+        dt = DiagnosticsToolbox(model)
+        dt.assert_no_structural_warnings()
+
+    @pytest.mark.component
+    def test_block_triangularization(self, SolEx_frame):
+        model = SolEx_frame
+        initializer = BlockTriangularizationInitializer(constraint_tolerance=2e-5)
+        initializer.initialize(model.fs.solex)
+
+        assert initializer.summary[model.fs.solex]["status"] == InitializationStatus.Ok
+
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_solve(self, SolEx_frame):
+        m = SolEx_frame
+        results = solver.solve(m)
+
+        # Check for optimal solution
+        assert check_optimal_termination(results)
+
+    @pytest.mark.integration
+    def test_solution(self, SolEx_frame):
```

Comment:
You should add a test using the diagnostics toolbox to `assert_no_numerical_warnings` here. I suspect this will likely show you a number of issues with this model that should be fixed.